### PR TITLE
ui/colorizer: rebase on maintained fork

### PR DESCRIPTION
**Pull request by Donnerinoern**
# Description

~~Added a new option for colorizer which, when enabled, will attach colorizer to any buffer with a filetype set in autoAttach.filetypes. I also fixed the indentation on colorizer's DEFAULT_OPTIONS.~~

Rebased colorizer.nvim on a maintained fork and updated the options accordingly.

## Type of change

- New feature

## Checklist

- [x] My code follows the style and contributing guidelines of this project.
- [x] I ran Alejandra to format my code (`nix fmt`).
- [x] I have performed a self-review of my own code and tested it.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no new warnings.
- [ ] This change requires a documentation update.
- [ ] I have updated the documentation accordingly.

## Screenshots & Logs

image

### CODE REVIEW

**Comment by NotAShelf** [comment]
Looks good. Though, I've noticed that nvim-colorizer has been abandoned for a while (4 years at this point) with a [maintained fork available](https://github.com/NvChad/nvim-colorizer.lua) with more intuitive configuration options.

Could I ask you to rebase on the maintained fork? It'd save us the trouble of revisiting this plugin later just to change plugins.

**Comment by Donnerinoern** [comment]
> Looks good. Though, I've noticed that nvim-colorizer has been abandoned for a while (4 years at this point) with a [maintained fork available](https://github.com/NvChad/nvim-colorizer.lua) with more intuitive configuration options.
> 
> Could I ask you to rebase on the maintained fork? It'd save us the trouble of revisiting this plugin later just to change plugins.

Nice catch, didn't notice that myself. I was too focused on just adding to what was already there, without making any drastic changes.

I can do a rebase on the fork tomorrow morning.

**Comment by FrothyMarrow** [comment]
@horriblename The maintained fork of the plugin configures the filetypes with a table constructor for lists:

```lua
filetypes = {
  'css',
  'javascript',
  html = { mode = 'foreground'; }
}
```
Regarding this setup, I have noticed that you have an option to collapse a key value pair to just value  in `expToLua` by prefixing the key with an `@`. But I am not sure if that is its intended use case or it should be handled differently.

**Comment by horriblename** [comment]
I actually didn't add that, maybe it was @NotAShelf?

in any case, it might be a bit better to expose a function like this

```nix
luaTableWithSequence ["javascript" "java"] {html = {...}}
```

I'm not sure how to best expose it as an option though

in this particular case, it should be fine just writing everything as
```nix
{
  javascript = {} ;
  css = {};
  html = {mode =... ;} ;
} 
```

**Comment by Donnerinoern** [comment]
This newest solution is comparable to what @horriblename suggested. If there's anything you would like changed, let me know.

**Review by horriblename — APPROVED** [review]
LGTM

**Comment by FrothyMarrow** [comment]
@NotAShelf Most of the options for the colorizer are wrapped inside the `options` block. I imagine you named it so because it was passed to the default options during setup. I think naming them `defaultOptions` or `userDefaultOptions`would be more understandable, especially since we now also have `filetypes` as an option. Additionally, @Donnerinoern, I believe `filetypes` should be placed outside the `options` block.

Other than that, it looks good to me. 👍

**Comment by NotAShelf** [comment]
>  I believe filetypes should be placed outside the options block.
I believe this is a fair point. @Donnerinoern could you move filetypes outside options before I merge this?

**Comment by Donnerinoern** [comment]
> I believe this is a fair point. @Donnerinoern could you move filetypes outside options before I merge this?

Done.

**Comment by NotAShelf** [comment]
Will merge once the CI passes.

**Comment by NotAShelf** [comment]
Thanks!

**Pull request merged**
